### PR TITLE
Add European Daylight Saving Time / Summertime

### PR DESCRIPTION
**Pull request by zevero**
I added 4 lines of code for European Daylight Saving Time

### CODE REVIEW

**Comment by andreiciocanvlad** [comment]
does this automatically correct the time ?

**Comment by zevero** [comment]
yes ... I use it for several projects

**Comment by andreiciocanvlad** [comment]
i`ll try to update my library compile and upload, it should change the time.

**Comment by andreiciocanvlad** [comment]
should i use DateTime::summertime_EU(0) in setup() ?

**Comment by hiddenotebook** [comment]
Nice thank you! 
Do you know that I can do to make the time change on winter/summer?

**Comment by zevero** [comment]
I may not understand your question, but usage is easy. Just set

    RTC_DS1307::use_summertime_EU = true;

and an hour will be added if we have summertime.

File: RTClib.h [hunk 1]
```diff
@@ -28,6 +28,7 @@ class DateTime {
     // 32-bit times as seconds since 1/1/1970
     uint32_t unixtime(void) const;
 
+    bool summertime_EU(uint8_t);
```
Comment (consolacion) [inline RTClib.h]: i suppose you mean bool use_summertime_EU(uint8_t);

**Comment by consolacion** [comment]
Do I understand that you have to set the various paremeters in the library itself such as 
tz and 
RTC_DS1307::use_summertime_EU =true?
or should i set that from within the sketch?

**Comment by zevero** [comment]
You should not have to edit anything in the library itself. use_summertime_EU is static public

**Comment by consolacion** [comment]
Thank you. it is just that RTC_DS1307::use_summertime_EU=true doesnt look like anything that  one would see in a sketch

File: RTClib.cpp [hunk 2]
```diff
@@ -179,6 +179,16 @@ long DateTime::secondstime(void) const {
   return t;
 }
 
+bool DateTime::summertime_EU(uint8_t tzHours) {
+// European Daylight Savings Time calculation by "jurs" for German Arduino Forum
+// input parameter: tzHours (0=UTC, 1=MEZ)
+// return value: returns true during Daylight Saving Time, false otherwise
+  if (m<3 || m>10) return false; // no summertime in Jan, Feb, Nov, Dez
+  if (m>3 && m<10) return true; // summertime in Apr, Mai, Jun, Jul, Aug, Sep
+  return (m==3  && (hh + 24 * d)>=(1 + tzHours + 24*(31 - (5 * (2000+yOff) /4 + 4) % 7)) || //spring
```
Comment (edgar-bonet) [inline RTClib.cpp]: `(5 * (2000+yOff) /4 + 4) % 7` can be simplified to `(5*yOff/4 + 5) % 7`.

File: RTClib.cpp [hunk 3]
```diff
@@ -179,6 +179,16 @@ long DateTime::secondstime(void) const {
...
+          m==10 && (hh + 24 * d)< (1 + tzHours + 24*(31 - (5 * (2000+yOff) /4 + 1) % 7))); //autumn
```
Comment (edgar-bonet) [inline RTClib.cpp]: `(5 * (2000+yOff) /4 + 1) % 7` can be simplified to `(5*yOff/4 + 2) % 7`.

File: RTClib.cpp [hunk 4]
```diff
@@ -262,9 +274,12 @@ DateTime RTC_DS1307::now() {
   uint8_t m = bcd2bin(WIRE._I2C_READ());
   uint16_t y = bcd2bin(WIRE._I2C_READ()) + 2000;
   
-  return DateTime (y, m, d, hh, mm, ss);
+  DateTime dt = DateTime(y, m, d, hh, mm, ss);
+  if (!use_summertime_EU || !dt.summertime_EU(0)) return dt;
```
Comment (edgar-bonet) [inline RTClib.cpp]: `summertime_EU(0)` implies [Western European Time](https://en.wikipedia.org/wiki/Western_European_Time), i.e. UTC+00:00/UTC+01:00 (UK, Ireland and Portugal). Most of Europe uses Central European, Eastern European or Moscow time.

**Comment by drak7** [comment]
If you're still interested in submitting this, please rebase on top of the latest library version and submit a new PR. Thanks!

**Pull request closed without merging**